Feature/responsive base Make Navbar Responsive in base.html

Gg1803 opened the pull request:
Updated the layout in base.html to make the navbar responsive using Tailwind CSS.

- Navbar now adjusts for mobile and desktop screens
- Ensures cleaner alignment and spacing
- Located in app/templates/base.html

chloetychang commented:
Hey Gargi!

Before I dive into a full review, could you clarify why so many additional requirements are being added to the virtual environment just to make the navigation bar responsive? This should be achievable using Tailwind alone, without needing extra dependencies.

Also, can we make sure to remove .DS_Store files before committing?

Lastly, it would be good to keep changes grouped logically — either move the welcome page changes to #71, or combine them into this PR along with the navbar update. As mentioned earlier, each PR should ideally focus on a single feature or a related set of changes to avoid confusion and merge conflicts.

Thanks Gargi!

chloetychang commented:
<img width="289" alt="Screenshot 2025-05-11 at 10 34 46 pm" src="https://github.com/user-attachments/assets/14d60285-cc44-4a22-a705-7e627cc986f0" />

chloetychang reviewed (changes requested):
See comments above for more detail.

Essential changes:
1. Remove unnecessary dependencies from requirements.txt to keep the virtual environment clean and minimal.
2. Remove .DS_Store system files.

Whether you want to feature welcome.html changes on this pull request, or under a separate pull request is your choice.

Thanks Gargi!

Gg1803 commented:
Hi @chloetychang,

Thanks for the detailed feedback!

I’d like to clarify a few things based on your comments:

✅ No additional dependencies were added by me
I haven’t modified requirements.txt or installed any new packages related to making the navigation bar responsive. All updates were implemented purely using Tailwind CSS utility classes. If any new entries appeared in the file, they may have resulted from another PR or system-level cache. I’ve attached a screenshot of my environment (black background) to show that I’m working within the existing virtual environment without introducing external libraries.

✅ .DS_Store files have been removed
I’ve cleaned up all .DS_Store files from the project and added them to .gitignore to prevent future commits. This has already been pushed.

✅ PR scope
Totally agree on keeping PRs logically grouped. Since this PR already includes the nav bar improvements, I’ll keep welcome.html changes separate under #71, as you suggested earlier — happy to maintain a clean separation of concerns.

Please let me know if you'd like me to take any further action. Appreciate your time and guidance!

Thanks again,
Gargi


<img width="171" alt="ss" src="https://github.com/user-attachments/assets/36706966-2093-4890-a428-3261051dbcc0" />
